.github/workflows/release.yml

```diff
@@ -0,0 +1,139 @@
+name: ezkl-lib-release
+on:
```

Comment:
We should use workflow_dispatch instead

This way it helps prevent tag collisions and also adds a separate step before releasing to make sure that at least a core contributor knows what they’re releasing.

`on:
  # Requires someone to trigger this build script via github workflows
  # It is required that the codebase is audited by contributors
  # before triggering to reduce the likelihood of supply chain attacks
  workflow_dispatch: `

Reply:
It's possible to gate builds to tagged versions with this check

```yaml
    if: "startsWith(github.ref, 'refs/tags/')"

```